src/utils/merkle.ts

```diff
@@ -31,7 +32,14 @@ export class MerkleTree {
   }
 
   static hash(a: string, b: string) {
-    const [aSorted, bSorted] = [a, b].sort();
+    let aSorted = a;
+    let bSorted = b;
+
+    if (toBN(aSorted) > toBN(bSorted)) {
+      aSorted = b;
+      bSorted = a;
+    }
+
```

Comment:
Nitpick suggestions
```suggestion
const [aSorted, bSorted] = [toBN(a), toBN(b)].sort((x: any, y: any) => x - y);
```

or in one line

```
  static hash = (a: string, b: string) =>
    pedersen(toBN(a) < toBN(b) ? [toBN(a), toBN(b)] : [toBN(b), toBN(a)]);
```

Reply:
I think the first one is good, I adapted it slightly to ensure we avoid any type / casting weirdness. Also realized that even the `>` with BNs had failing edge cases so I fixed and added an additional test case.